### DISCUSSION TITLE
Audio Improvements, DSP software

### DISCUSSION THREAD

**Discussion by yankscally** [godotengine/godot-proposals#4075] [Engine Core]
Hello,

I think Godot is extremely powerful engine and would be my preferred way to make generatrors and effects for digital audio workstations (DAWs). The audio engine is very capable and I was able to make a nice looking drum machine (video shown below). Export size is extremely low, and control nodes make application and menu setups very intuitive and developer friendly. MIDI is already supported which I'm super happy about.

I was sad to realise that, I lack the knowledge to convert a Godot project into a VST, which is the standardised license and filetype to use these types of applications. I think it would be awesome if there was a way to do this as an export, or some way to integrate it into an audio software, somehow...

Also, clever fuctions for the audio engine that allow user friendly tempo (BPM) control, speech recognition, real time signal processing pitch correction (autotune), recording audio, simple way to draw waveforms, built in frequency, built in tone (a=440 for example), LFO. There are is a lot of potential to open up the sound engine and make very elegant designs for music and audio production.

_drumpad for the numpad:_
(url

https://user-images.githubusercontent.com/27631913/155044617-51fe0e3e-cfb9-4b87-95ed-36e257174afc.mp4

)

**Comment by Akosmo** [comment]
I came across [this](https://github.com/godotengine/godot-proposals/issues/6410) recently! Could be a great addition to the engine (I imagine it'd be hard to convince a PR for this to be merged though). There's another proposal for CLAP and VST export too.
I also heard VST3 is MIT, so that might be an option too.
